### ISSUE TITLE
UI integration for ProgressiveSync feature

### ISSUE THREAD

**Issue by alexymantha** [argoproj/argo-cd#14463]
# Summary

Integrate the progressive sync feature to the ArgoCD UI to be able to easily see the status and progress of a rolling sync.

# Motivation

Currently, there does not seem to be a way to see from the UI what the status and progression of a rolling sync are. The only thing we see is that the applications are OutOfSync which does not give a lot of visibility over what is happening and can be misleading to users.

# Proposal

**Option 1** - Add a new status indicating that a rolling sync is in progress

1. New status in the `Sync Status` field 
Example: <img width="1416" alt="image" src="https://github.com/argoproj/argo-cd/assets/1365231/68a1ff42-fbf4-4e1c-9380-0aacd7c29fe1">

2. Instead of changing the sync status, add a sync condition similar to the sync window
<img width="1668" alt="image" src="https://github.com/argoproj/argo-cd/assets/1365231/8c0a20f2-33a7-4461-82ac-ff02ad17989b">
  - A caveat is that the Sync Window warning does not show up on the Application card, this one should show up in order to have an overview of what Applications are waiting for a rolling sync

**Option 2** - Add a dashboard to see the progression of an ApplicationSet

I'm not sure how that would work or where that should be since we don't have a view for ApplicationSets but it would be great to have a page where we could have an overview of the Applications of an ApplicationSet separated by the steps defined in the RollingSync spec and see the progression through the steps.

Open to suggestions on this one.

Options 1 and 2 are not mutually exclusive and both would be useful. I think Option 1 has the most value and Option 2 is more of a nice-to-have

**Comment by crenshaw-dev** [comment]
There's ongoing [work](https://github.com/argoproj/argo-cd/pull/13530) to add a first-class ApplicationSet experience to the UI.

Until then, a good first iteration of this idea might be to build on top of @speedfl's ApplicationSet UI extension: https://medium.com/@geoffrey.muselli/extend-argocd-ui-b651261e1003

**Comment by alexymantha** [comment]
Agreed for Option 2, it will make sense in the ApplicationSet experience

I think Option 1 would still be valuable in any case, even with the ApplicationSet experience. It will be useful to know on a per-app basis if it is waiting for a rolling sync. 

I will take a look at the UI extension, thanks for the suggestion!

**Comment by crenshaw-dev** [comment]
I think that's fair. We've talked about making that top bar a "hook" point for UI extensions. I think this would be a good use case for that.

**Comment by wmgroot** [comment]
One other simple thing we've considered that would go a long way would be adding support to see the sync status of child Applications on their cards from the view that has the ApplicationSet resource. This would allow you to see if child Applications are OutOfSync, Syncing, Healthy, for all Applications of an AppSet at the same time.

When I looked into this, the tricky part is visually distinguishing the sync status of an Application resource from the "is this resource synced and up to date in the k8s API" status of each Application resource.

**Comment by keithchong** [comment]
I can look at this.
